### DISCUSSION TITLE
Session Playtime per IGNs via Primary Group Query

### DISCUSSION THREAD

**Discussion by Tiger48XC** [plan-player-analytics/Plan#3899] [Q&A]
Hello,

I'm looking to run a query with this plugin to track our staff's playtime. Currently, I have to open each staff member's playtime and navigate to sessions to get their total playtime for the day. I was wondering if I'm missing a way to do the following:

Looking for a Query to populate the session data between selected dates to an Excel.
1: Name
2: UUID
3: Primary Group Value
4-Ect: Session Playtime Data for dates. (Example: Column - 12/01/24 & Row - 1 hr 38 mins)
Final Column: Total Playtime of dates selected (Good for quick count of week/month playtime)

I see the capability of Primary group value & choosing dates. I see I can export to excel, but right now I'm not getting the data I need. If there is a way to run a query for what I am looking for, can I have some assistance? 

I really appreciate any help you can provide.

**Comment by AuroraLS3** [comment]
Query page supports Primary group as a filter. 'Make a Query' 

Results include a table that can export to excel. List of sessions is not included in export.

**Reply by Tiger48XC** [reply]
Hello again,

Yes, I am tracking Primary Group and dates are filter selections. I have pulled all of this into Excel before.
Is there currently no capability to be able to pull session times? If not, that is okay! With the power of Query, I would love to see this as a future capability.

Below is a screenshot of a makeshift Excel "export" that I look forward to utilizing should the option be developed.
Thanks for your time!

![image](https://github.com/user-attachments/assets/19f252a7-7f2a-4673-942a-0f2a1e7900b3)

**Reply by AuroraLS3** [reply]
Oh! Date columns.. I don't think that's going to happen since grouping by date has so many issues with timezone offset that I haven't solved yet.

You could generate this report by querying plan_sessions table directly, though I guess it needs joins with plan_sessions.user_id->plan_users.id
And then to the extension tables. See the database schema on the wiki for how the data is linked together
